### ISSUE TITLE
NewPipe doesn't work properly on HUAWEI P20 Pro

### ISSUE THREAD

**Issue by naziemiec** [TeamNewPipe/NewPipe#6836]
Hello! I use NewPipe since 2021 for dowoading videos. In July I switched from OPPO A91 to HUAWEI P20 Pro due to my previous phone having crappy slo-mo and cut EIS feature from camera. And came into one problem. NewPipe either freezes on loading and then crashes, either NewPipe does load, but when you go to any video then you click on download button, the app freezes after the downloading window appears and then crashes also. This didn't happen on my previous phone.

Phone: HUAWEI P20 Pro

Accurate phone model: HUAWEI CLT-L29

OS: Android 10 (EMUI 10.0.0)

**Comment by neoOpus** [comment]
This seems to be an issue with EMUI

**Comment by SubSi** [comment]
I also have problems on Huawei Mate 20. Download causes crash.
## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ DE
* __Content Language:__ de-DE
* __App Language:__ de_DE
* __Service:__ none
* __Version:__ 0.21.8
* __OS:__ Linux Android 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to create service us.shandian.giga.service.DownloadManagerService: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4655)
	at android.app.ActivityThread.access$2800(ActivityThread.java:251)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2319)
	at android.os.Handler.dispatchMessage(Handler.java:110)
	at android.os.Looper.loop(Looper.java:219)
	at android.app.ActivityThread.main(ActivityThread.java:8393)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.schabi.newpipe.streams.io.StoredFileHelper.toString()' on a null object reference
	at us.shandian.giga.service.DownloadManager.loadPendingMissions(DownloadManager.java:165)
	at us.shandian.giga.service.DownloadManager.<init>(DownloadManager.java:77)
	at us.shandian.giga.service.DownloadManagerService.onCreate(DownloadManagerService.java:142)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4641)
	... 8 more

```
</details>
<hr>

**Comment by mhmdanas** [comment]
@subsi that's https://github.com/TeamNewPipe/NewPipe/issues/6687; it should be fixed in 0.21.9. Please inform us if it is not.

@naziemiec can you check if you face the issue on 0.21.9? If so, could you post the crash log?
